### PR TITLE
Add helm chart for the operator and daemonset

### PR DESCRIPTION
**Pull request by almaslennikov**

### CODE REVIEW

File: .github/workflows/image-push-main.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+on:
+  push:
+    branches:
+      - main
+jobs:
+  image-build-push:
+    name: Image build and push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set repository as lower-case output variable
```
Comment (maze88) [inline .github/workflows/image-push-main.yml:11]: isn't the repo name already lowercase?

File: .github/workflows/chart-push-release.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    tags:
+      - v*
+jobs:
+  package-and-push-helm-chart:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+      
+      - name: update chart
```
Comment (maze88) [inline .github/workflows/chart-push-release.yaml:17]: i'd make the steps' names have a capital first letter (like in the workflow name, and checkout step's name).

File: .github/workflows/build-test-lint.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+name: "Build, Test, Lint"
+on: [push, pull_request]
+jobs:
+  build:
+    strategy:
+      matrix:
```
Comment (maze88) [inline .github/workflows/build-test-lint.yml:6]: this 1*1*1 matrix seems not necessary... unless it is prep for future multi-version support?

File: .github/workflows/build-test-lint.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+#      - name: Lint
+#        run: make lint
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.x
```
Comment (maze88) [inline .github/workflows/build-test-lint.yml:43]: lint & test jobs use static version of go (compared to variable in build matrix). probably minor, but keep it in mind...

File: .github/workflows/image-push-main.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker meta
+        id: docker_meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ steps.repo_name.outputs.repository }}
+      - name: Build and push container image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
```
Comment (maze88) [inline .github/workflows/image-push-main.yml:36]: `context` and `file` have sane [defaults](https://github.com/docker/build-push-action?tab=readme-ov-file#customizing), no need to explicitly set them here unless there are high chances of changes to them in near future.

File: .github/workflows/image-push-main.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          push: true
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ghcr.io/${{ steps.repo_name.outputs.repository }}:latest
+          labels: ${{ steps.docker_meta.outputs.labels }}
+          file: ./Dockerfile
+      - name: Build and push container image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
```
Comment (maze88) [inline .github/workflows/image-push-main.yml:46]: `context` has sane [default](https://github.com/docker/build-push-action?tab=readme-ov-file#customizing), no need to explicitly set it here unless there are high chances of changes to it in near future.

File: .github/workflows/image-push-release.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    tags:
+      - v*
+jobs:
+  image-build-push:
+    name: Image build and push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set repository as lower-case output variable
+        id: repo_name
+        run: echo ::set-output name=repository::$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
```
Comment (maze88) [inline .github/workflows/image-push-release.yml:13]: isn't the repo name already lowercase?

File: .github/workflows/image-push-release.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          images: ghcr.io/${{ steps.repo_name.outputs.repository }}
+          tags: |
+            type=ref,event=tag
+          flavor: |
+            latest=false
+      - name: Build and push operator image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
```
Comment (maze88) [inline .github/workflows/image-push-release.yml:40]: `context` and `file` have sane [defaults](https://github.com/docker/build-push-action?tab=readme-ov-file#customizing), no need to explicitly set them here unless there are high chances of changes to them in near future.

File: .github/workflows/image-push-release.yml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          images: ghcr.io/${{ steps.repo_name.outputs.repository }}-daemon
+          tags: |
+            type=ref,event=tag
+          flavor: |
+            latest=false
+      - name: Build and push config daemon image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
```
Comment (maze88) [inline .github/workflows/image-push-release.yml:59]: `context` and `file` have sane [defaults](https://github.com/docker/build-push-action?tab=readme-ov-file#customizing), no need to explicitly set them here unless there are high chances of changes to them in near future.

File: .github/workflows/build-test-lint.yml (hunk 3 above)
Comment (almaslennikov) [reply .github/workflows/build-test-lint.yml:6]: We do target amd64 and arm64 for release, maybe we might also need to test arm64 in the future part of CI

File: .github/workflows/image-push-main.yml (hunk 1 above)
Comment (almaslennikov) [reply .github/workflows/image-push-main.yml:11]: Yes but the organization name (Mellanox) is not

Comment (maze88) [reply .github/workflows/image-push-main.yml:11]: ahh, that's right!

File: deployment/nic-configuration-operator-chart/templates/operator.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      {{- include "nic-configuration-operator.selectorLabels" . | nindent 8 }}
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+    spec:
+      tolerations: {{- toYaml .Values.operator.tolerations | nindent 8 }}
+      nodeSelector: {{- toYaml .Values.operator.nodeSelector | nindent 8 }}
+      affinity: {{- toYaml .Values.operator.affinity | nindent 8 }}
+      imagePullSecrets: {{ .Values.imagePullSecrets | default list | toJson }}
+      securityContext:
+        runAsNonRoot: true
```
[the remark this answers is not in the document]
Comment (almaslennikov) [reply deployment/nic-configuration-operator-chart/templates/operator.yaml:30]: We don't, it's runAsNonRoot

File: Makefile [hunk 11]
```diff
@@ -240,6 +241,24 @@ envtest: $(ENVTEST) ## Download setup-envtest locally if necessary.
 $(ENVTEST): $(LOCALBIN)
 	$(call go-install-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest,$(ENVTEST_VERSION))
 
+YQ := $(abspath $(LOCALBIN)/yq)
```
[the remark this answers is not in the document]
Comment (e0ne) [reply Makefile:244]: it's using in hack/release/chart-update.sh

File: deployment/nic-configuration-operator-chart/templates/operator.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (e0ne) [reply deployment/nic-configuration-operator-chart/templates/operator.yaml:30]: oh.. my bad, sorry

**Review by e0ne — APPROVED** [review]

**Pull request merged**